src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -401,6 +402,14 @@ public void Dispose()
             {
                 var exceptionAggregator = new ExceptionAggregator(_logger, "Failure to dispose landlord");
 
+                // we don't want to wake up database during dispose.
+                Parallel.ForEach(_wakeupTimers, (timer) =>
```

Comment:
Timer.Dispose is blocking, I think that you should call it serially, have an array of all MREs and then call WaitHandle.WaitAll(mreArray);